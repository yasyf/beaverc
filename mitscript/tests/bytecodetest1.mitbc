function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [None],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_const	0
				return
			]
		},
		function
		{
			functions = [],
			constants = [None],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_const	0
				return
			]
		},
		function
		{
			functions = [],
			constants = [None],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_const	0
				return
			]
		},
		function
		{
			functions = [],
			constants = ["val: ", "d", 1, "x", 2, 0],
			parameter_count = 2,
			local_vars = [x, val],
			local_ref_vars = [],
			free_vars = [],
			names = [print, foo],
			instructions = 
			[
				load_local	0
				load_const	5
				gt
				if	2
				goto	21
				load_const	0
				load_local	1
				add
				load_const	1
				add
				load_const	2
				load_global	0
				call
				pop
				load_local	1
				load_const	3
				add
				load_local	0
				load_const	2
				sub
				load_const	4
				load_global	1
				call
				return
				goto	3
				load_local	1
				return
			]
		}
	],
	constants = [7, "", 5, 2, 1, 0],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [print, input, intcast, x, foo],
	instructions = 
	[
		load_func	0
		store_global	0
		load_func	1
		store_global	1
		load_func	2
		store_global	2
		load_const	0
		store_global	3
		load_func	3
		store_global	4
		load_const	1
		load_const	2
		load_const	3
		load_global	4
		call
		load_const	4
		load_global	0
		call
		pop
		load_const	5
		return
	]
}