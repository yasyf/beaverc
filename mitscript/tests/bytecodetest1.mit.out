val:
val: x
val: xx
val: xxx
val: xxxx
xxxxx
